firetower

```diff
@@ -195,24 +195,29 @@ function get_flag() {
 
 function should_preserve_scrollback() {
   for var in "$@" ; do
-    if [[ $var == --preserve-scrollback ]] ; then
+    if [[ $var == --preserve-scrollback ]] || [[ $var == -p ]] ; then
       return 0
     fi
   done
   return 1
 }
 
 function get_directory() {
-  for var in "$@"
-  do
-    if [[ $var == --directory=* ]] ; then
-      echo "$var" | cut -d '=' -f 2
-      return
-    fi
+  for (( i=1; i<=$#; i++)); do
+    j=$((i+1))
+     if ( [[ ${!i} == -d ]] && [[ ${!j} == * ]] ) ; then
```

Comment:
Since it's impossible for `${!j} == *` to evaluate to anything but true, you might want a different test flag. `man test` will list all the options, but I think you want "string is nonzero length", and check `[[ -n "$string" ]]